app/cdap/components/RecipeList/RecipeTableRow.tsx

```diff
@@ -165,13 +163,26 @@ export const RecipeTableRow = ({
 
   return (
     <>
-      <div className="grid-row" onClick={!showAllColumns ? selectRecipeHandler : undefined}>
+      <div
+        className="grid-row"
+        onClick={!showAllColumns && selectRecipeHandler}
+        data-testid={`${recipe.recipeName
+          .toLowerCase()
+          .split(' ')
+          .join('-')}-recipe-row`}
+      >
         <div>{recipe.recipeName}</div>
         <div>{recipe.recipeStepsCount}</div>
         {showAllColumns && <div>{recipe.description}</div>}
         <div>{format(recipe.updatedTimeMillis, TYPES.TIMESTAMP_MILLIS)}</div>
         {showActions && (
-          <span onClick={handleActionsClick}>
+          <span
+            onClick={handleActionsClick}
```

Comment:
events like click, change etc should be written only on buttons and clickable items for better accessibility
i.e. semantic HTML practices

plz update across the PR